src/details/ArborX_DetailsConcepts.hpp

```diff
@@ -30,6 +30,33 @@ namespace ArborX
 namespace Details
 {
 
+struct not_a_type
+{
+  not_a_type() = delete;
+  ~not_a_type() = delete;
+  not_a_type(not_a_type const &) = delete;
+  void operator=(not_a_type const &) = delete;
+};
+
+// primary template handles all types not supporting the archetypal Op
+template <class, template <class...> class Op, class... Args>
+struct is_detected_impl : std::false_type
+{
+  using type = not_a_type;
```

Comment:
Are you using `type`?

Reply:
Not using it at the moment but useful if you want to make assertion on the return type.